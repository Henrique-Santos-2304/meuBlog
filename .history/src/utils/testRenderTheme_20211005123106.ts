import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
