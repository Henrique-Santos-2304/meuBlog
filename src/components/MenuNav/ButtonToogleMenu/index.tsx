import * as S from "./styles";
type props = {
  menuVisible: boolean;
};

const ButtonToogleMenu = ({ menuVisible }: props) => (
  <>
    {!menuVisible ? (
      <S.IconOpen
        role="button"
        aria-label="Botão Para abrir o Menu"
        title="Abrir Menu"
      ></S.IconOpen>
    ) : (
      <S.IconClose
        role="button"
        aria-label="Botão Para fechar o Menu"
        title="Fechar Menu"
      ></S.IconClose>
    )}
  </>
);
export default ButtonToogleMenu;
