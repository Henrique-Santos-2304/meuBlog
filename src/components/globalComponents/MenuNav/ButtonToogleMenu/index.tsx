import * as S from "./styles";
import { OpenClose } from "graphql/typesQueries/types";

type props = {
  menuVisible: boolean;
  openMenu: OpenClose;
  closeMenu: OpenClose;
};

const ButtonToogleMenu = ({ menuVisible, openMenu, closeMenu }: props) => {
  return (
    <>
      {!menuVisible ? (
        <S.IconOpen
          role="button"
          aria-label={openMenu.ariaLabel}
          title={openMenu.title}
        ></S.IconOpen>
      ) : (
        <S.IconClose
          role="button"
          aria-label={closeMenu.ariaLabel}
          title={closeMenu.title}
        ></S.IconClose>
      )}
    </>
  );
};
export default ButtonToogleMenu;
