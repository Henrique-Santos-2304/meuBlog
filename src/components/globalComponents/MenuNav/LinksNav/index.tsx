/* eslint-disable @typescript-eslint/no-non-null-assertion */
import * as S from "./styles";
import Link from "next/link";
import { linksNagivation } from "graphql/typesQueries/types";

type propsNavigation = {
  navigations: linksNagivation[];
};

const LinkNav = ({ navigations }: propsNavigation) => (
  <>
    {navigations!.map((nav) => (
      <S.ContentLink key={nav.title}>
        <Link href={nav.url}>{nav.text}</Link>
      </S.ContentLink>
    ))}
  </>
);

export default LinkNav;
