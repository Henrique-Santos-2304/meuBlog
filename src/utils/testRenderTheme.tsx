import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
