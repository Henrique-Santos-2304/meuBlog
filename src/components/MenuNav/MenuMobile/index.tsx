import * as S from "./styles";
import Image from "next/image";
import { useState } from "react";
import LinksNav from "../LinksNav";
import ButtonToogleMenu from "../ButtonToogleMenu";

const MenuNav = () => {
  const [menuVisible, setMenuVisible] = useState(false);
  return (
    <>
      <S.Contenticon
        onClick={() => setMenuVisible(!menuVisible)}
        aria-label="Botão de abrir e fechar menu de navegção"
      >
        <ButtonToogleMenu menuVisible={menuVisible} />
      </S.Contenticon>
      <S.Wrapper isVisible={menuVisible} aria-label="Menu de nagação de telas">
        <S.MenuNav
          isVisible={menuVisible}
          onClick={() => setMenuVisible(false)}
        >
          <S.PhotoContainer>
            <Image
              src="https://res.cloudinary.com/defnibbpl/image/upload/v1635012439/image_User_940e9f11e7.jpg"
              layout="fill"
            />
          </S.PhotoContainer>

          <LinksNav />
        </S.MenuNav>
      </S.Wrapper>
    </>
  );
};

export default MenuNav;
