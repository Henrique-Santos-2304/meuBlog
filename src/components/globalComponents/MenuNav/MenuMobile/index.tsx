/* eslint-disable @typescript-eslint/no-non-null-assertion */
import * as S from "./styles";
import Image from "next/image";
import { useState } from "react";
import LinksNav from "../LinksNav";
import ButtonToogleMenu from "../ButtonToogleMenu";
import { menuNav } from "graphql/typesQueries/types";

type propsMenu = {
  menuNav: menuNav;
};
const MenuNav = ({ menuNav }: propsMenu) => {
  const [menuVisible, setMenuVisible] = useState(false);

  return (
    <>
      <S.Contenticon
        onClick={() => setMenuVisible(!menuVisible)}
        aria-label={menuNav.ariaLabe}
        isVisible={menuVisible}
      >
        <ButtonToogleMenu
          menuVisible={menuVisible}
          openMenu={menuNav.openMenuNav}
          closeMenu={menuNav.closeMenuNav}
        />
      </S.Contenticon>
      <S.ContainerFull
        isVisible={menuVisible}
        onClick={() => setMenuVisible(false)}
        aria-label={menuNav.navigationPage.ariaLabel}
      >
        <S.Wrapper isVisible={menuVisible} data-testid="subMenuNav">
          <S.MenuNav isVisible={menuVisible}>
            <S.PhotoContainer>
              <Image
                src={menuNav.navigationPage.imageNav.url}
                alt={menuNav.navigationPage.imageNav.alt}
                layout="fill"
              />
            </S.PhotoContainer>

            <LinksNav navigations={menuNav.linksNagivation} />
          </S.MenuNav>
        </S.Wrapper>
      </S.ContainerFull>
    </>
  );
};

export default MenuNav;
