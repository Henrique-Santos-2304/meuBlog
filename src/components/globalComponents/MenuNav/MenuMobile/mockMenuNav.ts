import { mockButtonTooglrMenu } from "../ButtonToogleMenu/mockButtonToogleMenu";
import { mocknavLinks } from "../LinksNav/mockNaviLinks";

export const mockMenuNav = {
  menuNav: {
    openMenuNav: mockButtonTooglrMenu.openMenu,
    closeMenuNav: mockButtonTooglrMenu.closeMenu,
    linksNagivation: mocknavLinks.navigations,
    navigationPage: {
      ariaLabel: "Menu de nagação de telas",
      imageNav: {
        url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695300/image_User_940e9f11e7_34563f5075_b3e5a88c9b.jpg",
        alt: "esse sou eu",
      },
    },

    ariaLabe: "Botão de abrir e fechar menu de navegação",
  },
};
