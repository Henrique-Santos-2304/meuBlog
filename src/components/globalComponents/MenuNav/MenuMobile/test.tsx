/* eslint-disable @typescript-eslint/no-non-null-assertion */
import MenuNav from ".";
import { fireEvent, screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import { mockMenuNav } from "./mockMenuNav";

describe("<MenuNav />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<MenuNav {...mockMenuNav} />);
    expect(container).toMatchSnapshot();
  });
  it("should have items of the navigation menu ", () => {
    renderTheme(<MenuNav {...mockMenuNav} />);

    const buttonToogleMenu = screen.getByLabelText(
      /Botão de abrir e fechar menu de navegação/i
    );
    const navigation = screen.getByRole("navigation");
    const contentNavigation = screen.getByLabelText(
      /Menu de nagação de telas/i
    );
    expect(buttonToogleMenu).toBeInTheDocument();
    expect(navigation).toBeInTheDocument();
    expect(contentNavigation).toBeInTheDocument();
  });
  it("should handle menu with click in the button ", () => {
    renderTheme(<MenuNav {...mockMenuNav} />);

    const buttonToogleMenu = screen.getByLabelText(
      /Botão de abrir e fechar menu de navegação/i
    );
    const navigation = screen.queryByRole("navigation");
    const contentNavigation = screen.getByLabelText(
      /Menu de nagação de telas/i
    );

    expect(contentNavigation).toHaveStyle({ width: "35rem", opacity: "0" });

    expect(navigation).toHaveStyle({ opacity: "0" });
    expect(buttonToogleMenu).toHaveStyle({ background: "rgba(0, 0, 0, 0.65)" });

    fireEvent.click(buttonToogleMenu);
    expect(contentNavigation).toHaveStyle({ width: "100vw", opacity: "1" });
    expect(navigation).toHaveStyle({ opacity: "1" });
    expect(buttonToogleMenu).toHaveStyle({ background: "transparent" });

    fireEvent.click(contentNavigation!);
    expect(contentNavigation).toHaveStyle({ width: "35rem", opacity: "0" });
    expect(navigation).toHaveStyle({ opacity: "0" });
    expect(buttonToogleMenu).toHaveStyle({ background: "rgba(0, 0, 0, 0.65)" });
  });
});
