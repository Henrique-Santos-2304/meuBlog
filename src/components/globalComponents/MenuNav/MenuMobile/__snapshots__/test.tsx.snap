// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuNav /> should render  1`] = `
.c3 {
  width: 100%;
  max-width: 32rem;
  height: 76vh;
  -webkit-transition: all 0.6s ease-in-out;
  transition: all 0.6s ease-in-out;
  z-index: 15;
  background: rgba(0,0,0,0.91);
  border-radius: 2rem 0 0 2rem;
  border: 0.5rem inset #486afe;
  border-right: 0;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  opacity: 0;
}

.c5 {
  width: 10rem;
  height: 10rem;
  border-radius: 50%;
  margin: 1.3rem 0 2rem;
  position: relative;
  border-radius: 50%;
  border: 0.2rem groove #486afe;
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
  pointer-events: none;
}

.c5 img {
  border-radius: 50%;
}

.c5:hover {
  -webkit-transform: translate(0,-10%);
  -ms-transform: translate(0,-10%);
  transform: translate(0,-10%);
}

.c4 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  background: rgba(0,0,0,0.65);
  padding: 0.3rem;
  position: fixed;
  top: 2.6rem;
  right: 3rem;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 11;
  cursor: pointer;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c0:hover {
  border: 0.1rem solid #f8f8f8;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c2 {
  background: rgba(0,0,0,0.65);
  height: 100vh;
  width: 35rem;
  position: fixed;
  top: 0;
  right: 0;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
}

.c6 {
  color: #f8f8f8;
  font-size: 1.6rem;
  margin: 1.5rem 0;
  position: relative;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c6:hover::before {
  content: "";
  position: absolute;
  bottom: -1rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 120%;
  height: 0.3rem;
  background: #486afe;
  -webkit-animation: fSVfHE 0.6s ease-in-out;
  animation: fSVfHE 0.6s ease-in-out;
}

.c1 {
  color: #f8f8f8;
  font-size: 2.8rem;
}

@media (min-width:620px) {
  .c3 {
    border-radius: 2rem;
    border: 0.5rem solid #486afe;
    max-width: 40rem;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-left: 9rem;
  }
}

@media (min-width:620px) {
  .c0 {
    right: calc(50% - 30rem);
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

<div>
  <button
    aria-label="Botão de abrir e fechar menu de navegação"
    class="c0"
  >
    <svg
      aria-label="botão para abrir o menu"
      class="c1"
      fill="currentColor"
      height="1em"
      role="button"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Abrir Menu
      </title>
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </button>
  <div
    aria-label="Menu de nagação de telas"
    class="c2"
  >
    <nav
      class="c3"
      data-testid="subMenuNav"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="esse sou eu"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </div>
        <div
          class="c6"
        >
          <a
            href="/#"
          >
            Inicio
          </a>
        </div>
        <div
          class="c6"
        >
          <a
            href="/#projects"
          >
            Projetos
          </a>
        </div>
        <div
          class="c6"
        >
          <a
            href="/#skilss"
          >
            Habilidades
          </a>
        </div>
        <div
          class="c6"
        >
          <a
            href="/#contact"
          >
            Contato
          </a>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
